src/main/kotlin/no/nav/syfo/behandler/kafka/dialogmeldingfrombehandler/KafkaDialogmeldingFromBehandler.kt

```diff
@@ -84,3 +103,52 @@ private fun getPartnerId(fellesformatXML: String, navLogid: String, msgId: Strin
         null
     }
 }
+
+private fun getIdenterForBehandler(
+    fellesformatXML: String,
+    navLogid: String,
+    msgId: String,
+): Map<BehandleridentType, String> {
+    return try {
+        val xmlBehandler = getObjectFromXmlString<XMLHealthcareProfessional>(
+            fellesformatXML,
+            "HealthcareProfessional"
+        )
+        val behandlerIdenter = xmlBehandler.ident
+
+        behandlerIdenter.associate {
+            BehandleridentType.valueOf(it.typeId.v) to it.id
+        }
+    } catch (e: Exception) {
+        log.warn("Can't find behandleridenter in dialogmelding from behandler: navLogId: $navLogid, msgId: $msgId", e)
+        emptyMap()
+    }
+}
+
+private fun updateDialogmeldingEnabledForKontor(
+    partnerId: PartnerId,
+    database: DatabaseInterface,
+) {
+    val dialogmeldingEnabled = database.updateBehandlerKontorDialogmeldingEnabled(partnerId)
+
+    if (dialogmeldingEnabled)
+        log.info("Behandlerkontor with partnerId: $partnerId is ready for dialogmelding")
+    else
+        log.info("No behandlerkontor to update for $partnerId")
+}
+
+private fun updateIdenterForBehandler(
+    identer: Map<BehandleridentType, String>,
+    partnerId: PartnerId,
+    database: DatabaseInterface,
+) {
+    val behandlerFnr = identer[BehandleridentType.FNR]
```

Comment:
Nå får vi behandlerFnr fra HealthcareProfessional, men når vi skal bruke det til å finne en matchende behandler i databasen burde vi kanskje bruke `avsenderFnrFraDigSignatur` i `MottakenhetBlokk` i stedet? Eller er det kanskje greit å anta at identene inne i meldingen hører sammen, selv om det kanskje er noen andre som har signert?

Reply:
Det er vel behandlerFnr fra HealthcareProfessional som er den egentlige avsenderen (og det er for denne vi faktisk får disse andre identifiaktorene), så jeg synes dette er riktig som det er.